variable/test/variable_test.cpp

```diff
@@ -1106,6 +1106,18 @@ TEST(VariableTest, construct_mult_dev_unit) {
   EXPECT_EQ(int32_t(1) * units::kg, refMult);
 }
 
+TEST(VariableTest, datetime_dtype) {
+  auto dt =
+      makeVariable<scipp::core::time_point>(Values{scipp::core::time_point{}});
+  EXPECT_EQ(dt.dtype(), dtype<scipp::core::time_point>);
+}
+
+TEST(VariableTest, construct_time_unit) {
+  Variable refMult =
+      makeVariable<int64_t>(Dims(), Shape(), units::ns, Values{1000});
+  EXPECT_EQ(int64_t(1000) * units::ns, refMult);
```

Comment:
Python `int` is flexible-sized and accommodate any size number (within memory constraints).

```
PYBIND11_MODULE(python_example, m) {
    m.def("add", [](int64_t i, int64_t j) { return i + j; });
}
```
works as expected with any size of `int` on Windows as well
```
Python 3.7.7 (default, Apr 15 2020, 05:09:04) [MSC v.1916 64 bit (AMD64)] :: Anaconda, Inc. on win32
>>> import python_example as pe
>>> pe.add(2,1)
3
>>> pe.add(9999999999, 1)
10000000000
```

Reply:
Yes, but that is because Python11 does automatic conversions. Try the same with scipp variables and it will not work: Creating a variable with small ints on Windows gives int32_t, I think. Try:
```python
import scipp as sc
var = sc.Variable(dims='x', values=[1,2,3])
print(var.dtype)
```
Now if you pass that to a function we do *not* have implicit conversion from a variable with 32-bit ints to one with 64-bit ints.

Reply:
Sure, but simple arithmetic operations work as expected. The RHS "small int" `int32` gets properly cast into `int64` dtype.
```
>>> var = sc.Variable(dims='x', values=[1,2,3], dtype='int64')
>>> var+1
    <scipp.Variable>          int64      [dimensionless]  (x)  [2, 3, 4]
```

Reply:
Try without the `dtype` arg.

Reply:
The point is: Yes, we *can* construct with `int64_t` on Windows, but *once you have a variable* you do not get automatic conversion. In many cases users may not specify the dtype manually, so they would get different types on different platforms (whether this is good style is another question, and we should avoid this in *our* code, but we little control over user scripts).